### PR TITLE
Adding documentation about using agent only for metrics

### PR DESCRIPTION
**Pull request by eyalkoren**

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -426,7 +426,9 @@ If you want to dynamically change the status of the agent, use <<config-recordin
 [[config-instrument]]
 ==== `instrument` (added[1.0.0,Changing this value at runtime is possible since version 1.15.0])
 
-A boolean specifying if the agent should instrument the application to collect performance metrics for the app. When set to false, Elastic APM will not affect your application at all.
+A boolean specifying if the agent should instrument the application to collect traces for the app.
+ When set to `false`, Elastic APM will not apply bytecode instrumentation, thus it would have no effect on
```
Comment (felixbarny) [inline docs/configuration.asciidoc:430]: Technically, the agent still does some bytecode instrumentation but only for things related to manual APIs, such as the public API, the OT bridge, and Micrometer.

File: docs/metrics.asciidoc [hunk 2]
```diff
@@ -503,3 +504,14 @@ Fields:
...
+
+[float]
+[[metrics-only-mode]]
+=== Using the agent only for metrics collection
+
+There are cases where you would want to use the agent only to collect and ship metrics, without tracing any Java code.
+This may be useful, for example, when you already have an APM Server set up, and you want to ingest all metrics listed
+above from a JVM that runs something other than your application into Elasticsearch.
+
+In such cases, you may set the <<config-instrument, `instrument`>> config option to `false`. By doing so, agent will
```
Comment (felixbarny) [inline docs/metrics.asciidoc:516]:
```suggestion
In such cases, you may set the <<config-instrument, `instrument`>> config option to `false`. By doing so, the agent will
```

File: docs/metrics.asciidoc [hunk 3]
```diff
@@ -503,3 +504,14 @@ Fields:
...
 * `${name}`: The value of `functionCounter.count()` (the delta when using `CountingMode.STEP`).
 --
+
+[float]
+[[metrics-only-mode]]
+=== Using the agent only for metrics collection
+
+There are cases where you would want to use the agent only to collect and ship metrics, without tracing any Java code.
+This may be useful, for example, when you already have an APM Server set up, and you want to ingest all metrics listed
+above from a JVM that runs something other than your application into Elasticsearch.
```
Comment (felixbarny) [inline docs/metrics.asciidoc:514]: [minor] I find that example a bit confusing. Especially this bit:

> runs something other than your application into Elasticsearch

3rd party applications may also be traced with the agent and you may just want metrics from a 1st party app.

Maybe just remove that sentence?

**Review by felixbarny — APPROVED** [review]
LGTM, just a minor/optional comment

File: docs/metrics.asciidoc (hunk 3 above)
Comment (eyalkoren) [reply docs/metrics.asciidoc:514]: OK

**Pull request merged**